### PR TITLE
bf luis:build support to republish luis app if users specify a new publishing mode

### LINKED ISSUE #1143: bf luis:build cannot republish luis app to direct version mode if there is no lu content changed
## Versions
4.12.0

## Describe the bug
bf luis:build cannot republish to direct version mode if there is no lu content changed.

## To Reproduce
1. Create a lu file
2. Publish it with production slot mode
3. Republish it with direct version mode
4. You will find the the application in luis portal is still in production slot mode rather than direct version mode

## Expected behavior
The application in luis portal should be direct version mode

## Additional context

[bug]

### PR DESCRIPTION
**Pull request by feich-ms**
Fix #1143. The main fix is to support republishing luis app if users specify a new publishing mode, e.g. users have first published a luis app with directionVersionPublish mode, now they can also publish the same lu content to other modes like Production slot or Staging slot. Before the fix, the republishing logic is completely controlled by content comparisons between current lu content and existing luis portal lu content which means users have no ways to republish to other modes if their lu contents have no changes.

### CODE REVIEW

**Review by boydc2014 — APPROVED** [review]

**Comment by feich-ms** [comment]
@munozemilio could you help to approve this. Composer need this fix to unblock their directVersionPublish feature.

File: packages/lu/src/parser/lubuild/builder.ts [hunk 1]
```diff
@@ -504,4 +507,21 @@ export class Builder {
       app.intents = filteredIntents
     }
   }
+
+  forcePublish(endpoints: any, versionId: string, directVersionPublish: boolean, isStaging: boolean) {
+    let forcePublish = false
+    if (endpoints !== undefined) {
+      if (directVersionPublish && !Object.keys(endpoints).find(version => version.includes(versionId))) {
+        forcePublish = true
```
Comment (munozemilio) [inline packages/lu/src/parser/lubuild/builder.ts:515]: if forcePublish is always true, maybe this can be reduced to only setting the variable once

**Review by munozemilio — APPROVED** [review]
CIL for a possible simplification

File: packages/lu/src/parser/lubuild/builder.ts (hunk 1 above)
Comment (feich-ms) [reply packages/lu/src/parser/lubuild/builder.ts:517]: Thanks for the good catch, I just updated the code piece and made only once variable setting.

**Pull request merged**